README.md

```diff
@@ -1,2 +1,418 @@
+  results: [],
+  query: '',
+  loading: false,
+  error: ''
+});
+export const Tasks = {
+  async findResults({ model }) {
+    const res = await this.call(fetch, 'www.google.com/search');
+    return res.items;
+  }
+};
+export const Logic = {
+  name: 'SearchResults',
+
+  @Cmd.batch
+  Search(ctx, query) {
+    return [
+      this.InitSearch(query),
+      this.DoSearch()
+    ];
+  },
+
+  @Cmd.execLatest
+  DoSearch() {
+    return Task
+      .create(Tasks.findResults)
+      .success(this.SetResultsList)
+      .fail(this.HandleSearchFail)
+  },
+
+  @Cmd.update
+  InitSearch(ctx, query) {
+    return { query, loading: true };
+  },
+
+  @Cmd.update
+  SetResultsList(ctx, results) {
+    return {
+      results,
+      loading: false
+    };
+  },
+
+  @Cmd.update
+  HandleSearchFail(ctx, err) {
+    return {
+      error: 'Something weird happened',
+      loading: false
+    };
+  }
+};
+export const View = ({ model }) => (
+  <div>
+    {model.loading && <div>Loading...</div>}
+    {model.error && <div>{model.error}</div>}
+    {model.results.map((x, i) => <div key={i}>{x}</div>)}
+  </div>
+)
+```
+**`@Cmd.batch`** is a command type aimed to execute other commands. `batch` commands should decide what will be done next and in what order. Returned value can be an array, a command or null/undefined.
+
+**`@Cmd.execLatest`**. There is also `@Cmd.execEvery`. It is a command for executing an async function. A command of this type should return a `Task` object, which defines what async function should be executed and what is success/fail handler commands. Success command will be executed with a value returned by async function, and fail command will be executed with an error object throwed by async function. Success and fail commands are optional (will be executed a no-op command instead of not defined handler). A task function gets the same arguments as the command function which created the task.
+
+**`this.call(fetch, 'www.google.com/search')`** in the task. This thing inspired by `redux-saga` and needed to achive two goals: (1) it creates a task cancellation point and (2) it makes easier to test a task (you can pass some testing `call` func in a context which will mock results of `call`s). If some task marked as `execLatest`, then only one async fucnction can be executing at one point in time.
+
+### All together
+```js
+// Main.js
+import { Cmd } from 'mangojuice-core';
+import * as SearchForm from './SearchForm';
+import * as SearchResults from './SearchResults';
+
+export const createModel = () => ({
+  form: SearchForm.createModel(),
+  results: SearchResults.createModel()
+});
+export const Logic = {
+  name: 'Main',
+
+  config({ nest }) {
+    return {
+      children: {
+        form: nest(this.HandleSearchForm(), SearchForm.Logic),
+        results: nest(null, SearchResults.Logic),
+      }
+    }
+  },
+
+  @Cmd.batch
+  HandleSearchForm({ model }, cmd) {
+    if (cmd.is(SearchForm.Logic.Search.Before)) {
```

Comment:
if like above comment, we could avoid de Logic namespace, just flattening all logic components into the Block itself, like `SearchForm.Search.Before`....but it may be too much black magic?